packages/did/src/Did.utils.ts

```diff
@@ -263,8 +264,53 @@ export function isVerificationKey(
 export function isEncryptionKey(
   key: Partial<NewDidKey | DidKey> & Pick<NewDidKey | DidKey, 'type'>
 ): key is NewDidEncryptionKey | DidEncryptionKey {
-  const keyType = key.type.toLowerCase()
-  return encryptionKeyTypes.some((kt) => kt === keyType)
+  return encryptionKeyTypes.some((kt) => kt === key.type)
+}
+
+export const verificationKeyTypeToChainKeyType: Record<
+  VerificationKeyType,
+  DidDidDetailsDidVerificationKey['type']
+> = {
+  sr25519: 'Sr25519',
+  ed25519: 'Ed25519',
+  ecdsa: 'Ecdsa',
+}
```

Comment:
I do like the explicit declarative mappings more than imperative case changes, and I want to preserve them. A difference though is the split for verification and encryption keys.

Meanwhile this also makes `getMultiSignatureTypeFromKeypairType` obsolete and allows removing the comments and knowledge about the blockchain being case-agnostic here.